File: day-1/challenge-2.rb
require "csv"
require "pry"

frequencies_reached = [0]
current_frequency = 0
continue = true
i = 0

while continue do
  puts i
  CSV.read("input-1.csv").each do |csv|
    current_frequency += csv[0].to_i
    if frequencies_reached.include?(current_frequency)
      continue = false
      break
    end
    frequencies_reached <<  current_frequency 
  end
  i += 1
end

puts current_frequency

File: day-1/challenge-2.R
library("data.table")
frequencies <- fread("input-1.csv", sep = "\n")
frequencies <- frequencies[,1]
frequenciesReached <- c(0)
currentFrequency <- 0
continue <- TRUE 
n <- 1
i <- 1
rows <- nrow(frequencies)

while(continue) {
  print(n)
  i <- i + 1
  for(frequency in 1:rows) {
    currentFrequency <- currentFrequency + frequencies[frequency]
    if(match(currentFrequency, frequenciesReached, nomatch=0) > 0) {
      continue <- FALSE
      break
    } 

    frequenciesReached[i] = currentFrequency
  }
  n <- n + 1
}

print(currentFrequency)
